File: services/api-gateway/internal/modules/cache/models/session.go
package cachemodels

import (
	"time"
)

var _ Cachable = &Session{}

type Session struct {
	ID     string    `json:"session_id"`
	UserID string    `json:"user_id"`
	Token  string    `json:"access_token"`
	Exp    time.Time `json:"exp"`

	db int
}

func NewSession(id, userID, token string, exp time.Time) Session {
	return Session{
		ID:     id,
		UserID: userID,
		Token:  token,
		Exp:    exp,
		db:     0,
	}
}

func (s Session) GetDB() int {
	return s.db
}

File: services/frontend/src/lib/user/session.ts
import { UserOptional } from "./user";

export type AuthData = {
  session_id: string;
  access_token: string;
  exp: number;
  user: UserOptional;
};

export type Session = {
  session_id: string;
  user_id: string;
  access_token: string;
  exp: number;
  selected_company_id: string;
};
